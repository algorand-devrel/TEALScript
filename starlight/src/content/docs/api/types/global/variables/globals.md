---
editUrl: false
next: false
prev: false
title: "globals"
---

> **`const`** **globals**: `Object`

## Type declaration

### assetCreateMinBalance

> **assetCreateMinBalance**: [`uint64`](../type-aliases/uint64.md)

### assetOptInMinBalance

> **assetOptInMinBalance**: [`uint64`](../type-aliases/uint64.md)

### callerApplicationAddress

> **callerApplicationAddress**: [`Address`](../classes/Address.md)

### callerApplicationID

> **callerApplicationID**: [`Application`](../classes/Application.md)

### creatorAddress

> **creatorAddress**: [`Address`](../classes/Address.md)

### currentApplicationAddress

> **currentApplicationAddress**: [`Address`](../classes/Address.md)

### currentApplicationID

> **currentApplicationID**: [`Application`](../classes/Application.md)

### genesisHash

> **genesisHash**: [`bytes32`](../type-aliases/bytes32.md)

### groupID

> **groupID**: [`bytes32`](../type-aliases/bytes32.md)

### groupSize

> **groupSize**: [`uint64`](../type-aliases/uint64.md)

### latestTimestamp

> **latestTimestamp**: [`uint64`](../type-aliases/uint64.md)

### logicSigVersion

> **logicSigVersion**: [`uint64`](../type-aliases/uint64.md)

### maxTxnLife

> **maxTxnLife**: [`uint64`](../type-aliases/uint64.md)

### minBalance

> **minBalance**: [`uint64`](../type-aliases/uint64.md)

### minTxnFee

> **minTxnFee**: [`uint64`](../type-aliases/uint64.md)

### opcodeBudget

> **opcodeBudget**: [`uint64`](../type-aliases/uint64.md)

### round

> **round**: [`uint64`](../type-aliases/uint64.md)

### zeroAddress

> **zeroAddress**: [`Address`](../classes/Address.md)

## Source

[types/global.d.ts:800](https://github.com/algorandfoundation/tealscript/blob/18ba30a9/types/global.d.ts#L800)
