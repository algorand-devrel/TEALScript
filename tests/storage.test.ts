/* eslint-disable func-names */
/* eslint-disable prefer-arrow-callback */
import { describe, test, expect } from '@jest/globals';
import { getMethodTeal, artifactsTest } from './common';

async function getTeal(methodName: string) {
  return getMethodTeal('tests/contracts/storage.algo.ts', 'StorageTest', methodName);
}

artifactsTest('StorageTest', 'tests/contracts/storage.algo.ts', 'tests/contracts/artifacts/', 'StorageTest');

const ops: {[type: string]: {[method: string]: string}} = {
  global: {
    Get: 'app_global_get',
    Put: 'app_global_put',
    Delete: 'app_global_del',
    Exists: 'app_global_get_ex',
  },
  local: {
    Get: 'app_local_get',
    Put: 'app_local_put',
    Delete: 'app_local_del',
    Exists: 'app_local_get_ex',
  },
  box: {
    Get: 'box_get',
    Put: 'box_put',
    Delete: 'box_del',
    Exists: 'box_get',
  },
};

['global', 'local', 'box'].forEach((storageType) => {
  ['Key', 'Map'].forEach((storageClass) => {
    describe(`${storageType}${storageClass}`, function () {
      ['Put', 'Get', 'Delete', 'Exists'].forEach((method) => {
        test(`${storageType}${storageClass}${method}`, async function () {
          const teal = await getTeal(`${storageType}${storageClass}${method}`);
          const expectedTeal: string[] = [];

          if (storageType === 'local') expectedTeal.push('frame_dig -1 // a: account');

          if (['local', 'global'].includes(storageType) && method === 'Exists') expectedTeal.push('txna Applications 0');

          expectedTeal.push('byte "foo"');

          if (method === 'Put') {
            expectedTeal.push('byte "bar"');
          }

          expectedTeal.push(ops[storageType][method]);

          if (method === 'Exists') {
            expectedTeal.push('swap');
            expectedTeal.push('pop');
          }

          if (method === 'Get') {
            if (storageType === 'box') expectedTeal.push('assert');
            expectedTeal.push('byte "bar"');
            expectedTeal.push('==');
            expectedTeal.push('assert');
          }

          expect(teal.slice(1)).toEqual(expectedTeal);
        });
      });
    });
  });
});
