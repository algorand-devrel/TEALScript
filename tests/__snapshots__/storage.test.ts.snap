// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Storage StorageTest Artifacts Maintains program size 1`] = `1220`;
